ad pydantic validation and account get, put api

Vghxv opened the pull request:
the account api is not done yet, it's demo for basic module function

Vghxv committed:
ad pydantic validation and account get, put api

PEKKA1117 commented on core/settings.py:
```diff
@@ -11,3 +11,4 @@
     "api_prefix": str(config["SERVER"]["api_prefix"]),
     "api_docs" : str(config["SERVER"]["api_docs"]),
 }
+Settings: Final
```
'Final' name should be initialized with a value

PEKKA1117 commented on model/account.py:
```diff
@@ -1,12 +1,17 @@
+    email: Optional[str] 
+    phone: Optional[str] 
+    credit_card: Optional[str] 
+    birthday: Optional[str] 
```
sussy spaces at EOL

PEKKA1117 commented on model/account.py:
```diff
@@ -1,12 +1,17 @@
+    birthday: Optional[str] 
+    address: Optional[str]
+    is_active: Optional[bool]
 
```
Extra blank line

PEKKA1117 commented on core/settings.py:
```diff
@@ -1,6 +1,6 @@
 import os
-
+from typing import Final
 config = configparser.ConfigParser()
```
a blank line to separate inclusion and code

PEKKA1117 commented on model/general.py:
```diff
@@ -0,0 +1,11 @@
+
+class ErrorModel(BaseModel):
+    status: str
+    msg: Optional[str] = None
```
Missing blank at EOF

PEKKA1117 commented on router/account.py:
```diff
@@ -2,17 +2,52 @@
 from fastapi import Depends
 from fastapi import HTTPException
```
Unused lib.

PEKKA1117 commented on router/account.py:
```diff
@@ -2,17 +2,52 @@
 from fastapi import HTTPException
 from fastapi import status
-from model.account import modelExample
+from fastapi import Request
```
Unused lib.

PEKKA1117 commented on router/account.py:
```diff
@@ -2,17 +2,52 @@
+        content=jsonable_encoder({
+                "data":result
+            })
 )
```
Sussy idents.

PEKKA1117 commented on router/account.py:
```diff
@@ -2,17 +2,52 @@
         status_code=status.HTTP_200_OK,
-        content={"status": "success", "message": "Hello World"}
+        content=jsonable_encoder({
+                "data":result
```
space after `:`

PEKKA1117 commented on router/account.py:
```diff
@@ -2,17 +2,52 @@
+    sql = f"""
+        UPDATE `Account` SET {sql_set_text} WHERE account_uuid = %s;
+    """
+    result = execute_query(sql,(sql_set_values+(account_form["account_uuid"],)))
```
Add spaces.

PEKKA1117 commented on router/account.py:
```diff
@@ -2,17 +2,52 @@
+        status_code=status.HTTP_200_OK,
+        content=jsonable_encoder({
+                "data":result
+            })
```
Sussy idents.

PEKKA1117 commented on utils/as_form.py:
```diff
@@ -0,0 +1,41 @@
+            instance = cls(**data)
+        except ValueError as e:
+            raise HTTPException(status_code=422, detail=str(e))
+        
```
Sussy spaces.

PEKKA1117 commented on utils/as_form.py:
```diff
@@ -0,0 +1,41 @@
+
+    # Update the function signature
+    sig = inspect.signature(as_form_func)
+    
```
Sussy spaces.

PEKKA1117 commented on utils/as_form.py:
```diff
@@ -0,0 +1,41 @@
+            default=inspect.Parameter.empty if is_required else None
+        )
+        new_parameters.append(parameter)
+    
```
Sussy spaces.

PEKKA1117 commented on utils/as_form.py:
```diff
@@ -0,0 +1,41 @@
+    
+    # Attach the as_form method to the Pydantic model
+    setattr(cls, 'as_form', as_form_func)
+    
```
Sussy spaces.

PEKKA1117 commented on utils/as_form.py:
```diff
@@ -0,0 +1,41 @@
+    
+    sig = sig.replace(parameters=new_parameters)
+    as_form_func.__signature__ = sig
+    
```
Sussy spaces.

PEKKA1117 commented on utils/as_form.py:
```diff
@@ -0,0 +1,41 @@
+    # Attach the as_form method to the Pydantic model
+    setattr(cls, 'as_form', as_form_func)
+    
+    return cls
```
Leave a blank line after EOF.

PEKKA1117 commented on utils/db_process.py:
```diff
@@ -0,0 +1,93 @@
+from typing import Union, Tuple, Dict, List
+from typing import Final
+import datetime
+load_dotenv()
```
A blank line to separate inclusion and code.

PEKKA1117 commented on utils/db_process.py:
```diff
@@ -0,0 +1,93 @@
+    "password": os.getenv("DB_PASSWORD"),
+    "database": os.getenv("DB_DATABASE"),
+}
+pool_config: Final
```
'Final' name should be initialized with a value. [Ref](https://peps.python.org/pep-0591/#syntax)

PEKKA1117 commented on utils/db_process.py:
```diff
@@ -0,0 +1,93 @@
+        else:
+            cursor.execute(sql, param)
+        result = cursor.fetchall()
+        
```
Sussy space.

PEKKA1117 commented on utils/db_process.py:
```diff
@@ -0,0 +1,93 @@
+        connection.close()
+        column_names = [desc[0] for desc in cursor.description]
+        result = [dict(zip(column_names, row)) for row in result]
+        return result
```
Expected type 'dict', got 'list[dict]' instead

PEKKA1117 commented on utils/db_process.py:
```diff
@@ -0,0 +1,93 @@
+    except mysql.connector.Error as e:
+        print(e)
+        cursor.close()
+        connection.close()
```
Vars can be not declared when there's something wrong to the connection. Tbh, I don't really know how Python's errors are handled(or the life scope thing).

PEKKA1117 commented on utils/db_process.py:
```diff
@@ -0,0 +1,93 @@
+        print(e)
+        cursor.close()
+        connection.close()
+        return False
```
Expected type 'dict', got 'bool' instead

PEKKA1117 commented on utils/db_process.py:
```diff
@@ -0,0 +1,93 @@
+        print(e)
+        cursor.close()
+        connection.rollback()
+        connection.close()
```
Vars can be not declared when there's something wrong to the connection. Tbh, I don't really know how Python's errors are handled(or the life scope thing).

PEKKA1117 commented on utils/db_process.py:
```diff
@@ -0,0 +1,93 @@
+        connection.rollback()
+        connection.close()
+        return False
+    
```
Sussy space.

PEKKA1117 commented on utils/db_process.py:
```diff
@@ -0,0 +1,93 @@
+        connection.close()
+        return False
+    
+def dict_to_sqltext(_dict: Dict, exclude_col: List[str] = []) -> Tuple[str, tuple]:
```
`dict_to_sql_command` would be a better name?

PEKKA1117 commented on utils/db_process.py:
```diff
@@ -0,0 +1,93 @@
+        return False
+    
+def dict_to_sqltext(_dict: Dict, exclude_col: List[str] = []) -> Tuple[str, tuple]:
+    sqltext = []
```
`sql_command` would be a better name?

PEKKA1117 commented on utils/db_process.py:
```diff
@@ -0,0 +1,93 @@
+        connection.close()
+        return False
+    
+def dict_to_sqltext(_dict: Dict, exclude_col: List[str] = []) -> Tuple[str, tuple]:
```
> If a mutable object (like a list, dictionary, or any instance of a user-defined class) is used as a default argument in a Python function, it can lead to unexpected behavior. That's because function argument defaults are evaluated only once at function definition time, not each time the function is called.
Here's an example to demonstrate this:
```python
def append_to(num, target=[]):
    target.append(num)
    return target

print(append_to(1))  # outputs: [1]
print(append_to(2))  # outputs: [1, 2], not [2] as you might expect
```

PEKKA1117 replied:
```python
def dict_to_sqltext(_dict: Dict, exclude_col: List[str] = None) -> Tuple[str, tuple]:
    if exclude_col is None:
        exclude_col = []
```
would do it

PEKKA1117 commented on utils/db_process.py:
```diff
@@ -0,0 +1,93 @@
+            for v_i in value:
+                dict_delete_none(v_i)
+
+    return _dict
```
Blank line at EOF.

Dragonfly911117 reviewed (changes requested):
I'll post a patch for most of my complains.

Dragonfly911117 committed:
Refactor code for readability and efficiency

This commit includes code refactoring across multiple files (model/general.py, model/account.py, utils/db_process.py, server.py, router/account.py, core/settings.py, utils/as_form.py, main.py, requirements.txt). Changes were made to how SQL queries are crafted and executed for improved readability and efficiency. Unused imports were removed & necessary ones reorganized. In addition, formatting and white spaces were cleaned up according to PEP8 standards for cleaner, more consistent code.

Vghxv committed:
Merge pull request #1 from Dragonfly911117/dev

Refactor code for readability and efficiency
